### DISCUSSION TITLE
title tag not there with SSR off, pre-render off

### DISCUSSION THREAD

**Discussion by realJogicodes** [sveltejs/kit#12239] [Q&A]
Hi

using ahref, I did an SEO audit for https://allboilerplates.com which is a dictionary app powered by svelte kit. The results were decent, almost no issues detected bar one: The route, where users can submit their boilerplate for listing has SSR and prerender off in the +page.server.ts file.

I am guessing because of that, there's no title tag displayed on the markup and ahref is flagging the page as having errors.

From memory, I needed to switch off prerender because there was an issue with the submission form otherwise causing a 500 on the server side and not rendering the form on the client and I think SSR if off for the same reason.

Anyway, is there a way to get a title tag when both prerender and SSR are false, or is that just one of those things I need to live with?

Cheers

**Comment by teemingc** [comment]
> From memory, I needed to switch off prerender because there was an issue with the submission form otherwise causing a 500 on the server side and not rendering the form on the client and I think SSR if off for the same reason.

Can you expand on this? Is the submission form using a SvelteKit action? Prerendering does not work with SvelteKit actions but SSR shouldn't be an issue.

> Anyway, is there a way to get a title tag when both prerender and SSR are false, or is that just one of those things I need to live with?

You can set the title tag in the `app.html` template, which would be used before SSR and CSR occur. The main issue is that without SSR, CSR only kicks much later so the title tag from the Svelte template gets rendered late. The solution is to either prerender or enable SSR so that the title tag gets rendered earlier for SEO.

**Reply by realJogicodes** [reply]
Yeah I thought so, thanks. I forgot why I switched prerendering and SSR off. The little I do remember is that I debugged something because the from would run on the server for some reason and SSR off fixed the issue but my memory is not great so that might be wrong.

It's been a while since I did that, maybe I can reenable prerender and SSR, that would fix it. I learned something valuable about svelte kit in the process though, understand better how it all works now

**Reply by teemingc** [reply]
Is the form posting to somewhere that’s not svelte kit?
